### PR TITLE
[gradle] New Compose Compiler gradle plugin checks.

### PR DESCRIPTION
**Pull request by terrakok**
If KGP >= 2.0.0-RC1 new Compose gradle plugin has to be applied.

To throw a configuration error if it's not.

### CODE REVIEW

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeExtension.kt
Comment (igordmn) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeExtension.kt:34]: As we still don't have compatible `org.jetbrains.kotlin.plugin.compose` plugin, we shouldn't deprecate it now.

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt
Comment (eymar) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt:37]: Some thoughts:
- RC1 is the only version we can test this chnage at the moment.
- RC1 is okay for android+jvm projects where we can test this change. 

But we should not force 1.6.10-beta01 users to apply a new compose-compiler-gradle plugin, because it doesn't support compose multiplatform yet.

Perhaps we don't merge this now, so it won't go into 1.6.10-beta01. 
Before merging, we can update to RC2 (hopefully) or RC3 version here.

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeExtension.kt
Comment (eymar) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeExtension.kt:34]: Maybe it's okay to keep this deprecation if we postpone merging this PR until kotlin 2.0.0-RC2.

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/internal/Version.kt
Comment (eymar) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/internal/Version.kt:16]: is it simple alphabetic comparison for 'meta'?

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeExtension.kt
Comment (igordmn) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeExtension.kt:34]: true

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt
Comment (igordmn) [reply gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt:40]: If not this, we'll need to put a link to the doc later (when it is ready)

**Comment by terrakok** [comment]
The error happens because there is no Kotlin 2.0.0-RC2 yet

File: gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt
Comment (igordmn) [inline gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeCompilerKotlinSupportPlugin.kt:42]:
```suggestion
            "you must apply \"$newComposeCompilerKotlinSupportPluginId\" plugin. See the migration guide https://www.jetbrains.com/help/kotlin-multiplatform-dev/compose-compiler/compose-compiler.html#migrating-a-compose-multiplatform-project"
```

A preliminary link to the guide (draft is [here](https://helpserver.labs.jb.gg/help/kotlin-multiplatform-dev/compose-compiler/compose-compiler.html)). Let's merge it in the PR, and revert/change if needed before `beta03`.

**Comment by igordmn** [comment]
> The error happens because there is no Kotlin 2.0.0-RC2 yet

Is it only in tests? Can we disable them? We need to merge the PR before it will be available

**Review by igordmn — APPROVED** [review]
Let's sync tomorrow before the merge.

**Pull request merged**
